Add Mobile-View for Livechat

HenrikScholl opened the pull request:
<!-- INSTRUCTION: Your Pull Request name should start with one of the following tags -->
<!-- [NEW] For new features -->
<!-- [FIX] For bug fixes -->
<!-- [BREAK] For pull requests including breaking changes -->

<!-- INSTRUCTION: Keep the line below to notify all core developers about this new PR -->
@RocketChat/core 

<!-- INSTRUCTION: Inform the issue number that this PR closes, or remove the line below -->
<!--Closes #ISSUE_NUMBER-->

<!-- INSTRUCTION: Tell us more about your PR with screen shots if you can -->

mrsimpson committed:
Merge branch 'develop' into feature/merge-0.55.1

ruKurz committed:
Merge pull request #16 from mrsimpson/feature/merge-0.55.1

Merging current Rocket.Chat 0.55.1

ruKurz committed:
#15 - Add advanced configuration options for piwik analytics.
- Removed unnecessary code for setting a custom tracking url
- Make the list of domains editable
- Changed comment

mrsimpson committed:
Merge pull request #18 from mrsimpson/feature/15-piwik-analytics

#15 - Add advanced configuration options for piwik analytics.

ruKurz committed:
#20 The CSS format for showing the knowledge base is broken.
- Fixed formatting issues according to stylelint

mrsimpson committed:
Merge branch 'develop' into bug/css-formatting

mrsimpson committed:
Merge pull request #21 from mrsimpson/bug/css-formatting

#20 The CSS format for showing the knowledge base is broken.

mrsimpson committed:
Support private expert channels

mrsimpson committed:
Merge branch 'hotfix/experts-channel'

An expert channel which was configured as private channel prevented startup

mrsimpson committed:
Merge branch 'hotfix/experts-channel' into develop

mrsimpson committed:
Merge remote branch 'develop' of https://github.com/mrsimpson/Rocket.Chat into develop

ruKurz committed:
Merge branch 'release/0.55.1-0.2'

Making release for beta test.

ruKurz committed:
Merge branch 'master' of https://github.com/mrsimpson/Rocket.Chat

ruKurz committed:
Merge branch 'master' of https://github.com/mrsimpson/Rocket.Chat into release/0.55.1-0.2

Merging hotfix from master into release branch.

ruKurz committed:
Merge branch 'release/0.55.1-0.2'

ruKurz committed:
Merge branch 'release/0.55.1-0.2' into develop

mrsimpson committed:
- hide standard features
- add new labels

mrsimpson committed:
Merge branch 'hotfix/left-hand-panel'

mrsimpson committed:
Fix remove expert users

mrsimpson committed:
Merge pull request #30 from mrsimpson/hotfix/left-hand-panel

Hotfix/left hand panel

mrsimpson committed:
Merge branch 'master' of https://github.com/mrsimpson/Rocket.Chat

mrsimpson committed:
changed request API

mrsimpson committed:
Hide experts-headline if authorization for creation of experts channels is missing

mrsimpson committed:
Incompatible change of Request API - fixes #32

mrsimpson committed:
Hide expertise-headline - fixes #31

ruKurz committed:
Merge pull request #33 from mrsimpson/hotfix-bundle/0.56.0-0.2.2

Hotfix bundle/0.56.0 0.2.2

Commit:
first version of assistify plugin

Commit:
Merge branch 'master' of https://github.com/mrsimpson/Rocket.Chat

Commit:
add webhook in order to enable asyncronous smarti analysis

Commit:
re-enable "close channel"

Commit:
add a comment for a missing auth part

mrsimpson committed:
Proxy smarti-UI on RocketChat server. Fixes #42

ruKurz committed:
#42 Proxy smarti-UI on RocketChat server.
- use script tag instead 'eval'

ruKurz committed:
Merge pull request #43 from mrsimpson/hotfix/smarti-script-proxy

Proxy smarti-UI on RocketChat server.

mrsimpson committed:
Merge branch 'develop' of https://github.com/mrsimpson/Rocket.Chat into develop

mrsimpson committed:
Merge branch 'test/smarti-0.1.0-pre' into develop

mrsimpson committed:
Change API to Rocket.Chat API V1 (authenticated methods). Fixes #32 (#49)

mrsimpson committed:
Minor functional enhancements (#50)

* Allow lowercase topic names
* Error handling Smarti not available : Do not create `undefined` script tags
* Help-request actions button shall disappear once the help-request has been closed (#38)
* Make requests for expertises respect case as well (#36)
* Remove subscription of owner who created a request on closing a help request (fixes #38)
* #47 - Closing a help request does not store the conversation correctly
- Added some debugging output
- Some formatting
* Do not lowercase experts channel
* Satisfy linter and remove console dependency

mrsimpson committed:
Harmonize behavior of live chats and requests with respect to actions handling
Fixes #44

mrsimpson committed:
Harmonize behavior of live chats and requests with respect to actions handling (#51)

Fixes #44

ruKurz committed:
#53 - Pass Solr-endpoint query suffix as options to the Smarti widget

ruKurz committed:
Fixes #55 - Disable keyword search per default

mrsimpson committed:
Harmonize behavior of live chats and requests with respect to actions handling
Fixes #44

Do not open Assistify-tab if running in the mobile app (fixes #41). However, a browser opening the mobile view is likely to open the panel on opening the room still.

mrsimpson committed:
Merge branch 'hotfix/#52-unsubscribed-room' into release/0.56.0-0.3.0

ruKurz committed:
Fixes #55 Disable keyword search per default
Fixes a small / not related lint Issue

ruKurz committed:
Merge pull request #54 from mrsimpson/feature/search-config

Search config

mrsimpson committed:
Merge remote-tracking branch 'origin/feature/search-config' into release/0.56.0-0.3.1

ruKurz committed:
Merge pull request #57 from mrsimpson/release/0.56.0-0.3.1

Release 0.56.0 0.3.1

Commit:
add configuration parameter for Assistify Widgets posting behaviour

ruKurz committed:
Merge branch 'develop' into master

ruKurz committed:
Merge pull request #59 from redlink-gmbh/master

add configuration parameter for Assistify Widgets posting behaviour

mrsimpson committed:
Upgraded Rocket.Chat to 0.57.2

Conflicts:
Expertise and request room types to be considered in files which have been converted from .coffee to .js:

- packages/rocketchat-ui/client/views/app/room.js
- packages/rocketchat-ui-admin/client/rooms/adminRooms.js

package.js merged

mrsimpson committed:
Upgraded Rocket.Chat to 0.57.2

Conflicts:
Expertise and request room types to be considered in files which have been converted from .coffee to .js:

- packages/rocketchat-ui/client/views/app/room.js
- packages/rocketchat-ui-admin/client/rooms/adminRooms.js

package.js merged

mrsimpson committed:
Provide labels for custom room types (as required per 0.57)

mrsimpson committed:
satisfy eslint

mrsimpson committed:
satisfy stylint

mrsimpson committed:
satisfy stylint - one more reverting what `stylelint "**/*.less" --fix` has done to the file previously...
Note to myself: report error in `stylelint --fix`

mrsimpson committed:
Error handling for invalid smarti configuration. Fixes #68

mrsimpson committed:
submit no username onMessage, but only the GUID

mrsimpson committed:
manual invalidation of smarti widget

mrsimpson committed:
automated daily invalidation of smarti widget

mrsimpson committed:
make dynamically loaded SmartiWidget debuggable

mrsimpson committed:
rename button "Hilfe beenden"

ruKurz committed:
#79 - Change the powered by logo in live-chat (#80)

mrsimpson committed:
added error and success message

ruKurz committed:
Merge pull request #78 from mrsimpson/feature/#77-smarti-widget-invalidation

#77 smarti widget invalidation

mrsimpson committed:
Dirty hack for being able to delete an expertise. Fixes #58
The actual correction would be to provide a real authorization object, but that lead to exceptions which could not be corrected so far.
Thus, this commit provides a short-term-low-risk-workaround.

ruKurz committed:
Merge remote-tracking branch 'upstream/master' into merge/v0.58.1

mrsimpson committed:
Make Smarti more robust with respect to trailing slashes provided or missing

mrsimpson committed:
Make Smarti more robust with respect to trailing slashes provided or missing - fix copy-paste-error

vickyokrm committed:
feature/#62-refresh-send-icon

vickyokrm committed:
feature/#62-refresh-send-icon

ruKurz committed:
Merge pull request #81 from mrsimpson/release/v0.58.1-0.3.2

Release Rocket Chat 0.58.1 Assistify 0.3.2

ruKurz committed:
Merge pull request #83 from mrsimpson/feature/#62-refresh-send-icon

Feature/#62 refresh send icon

mrsimpson committed:
Make fix for input selection more robust (#84)

* use Meteor's internal selector instead of a global jQuery

* Make the focus input event a bit more robust

* Undo accidental cut

* code style

mrsimpson committed:
Minor fix for message box focusing (#85)

* feature/#62-refresh-send-icon

* feature/#62-refresh-send-icon

* Make fix for input selection more robust (#84)

* use Meteor's internal selector instead of a global jQuery

* Make the focus input event a bit more robust

* Undo accidental cut

* code style

* satisfy esLint

mrsimpson committed:
fixes #88

mrsimpson committed:
Merge pull request #89 from mrsimpson/hotfix/#88-users-expertise-channels

Allow user management in expert channels

mrsimpson committed:
Merge branch 'hotfix/experts-channel-creation' into hotfix/automated-experts-channel

mrsimpson committed:
An empty array is still an array...

ruKurz committed:
Merge pull request #1 from assistify/hotfix/automated-experts-channel

Create an experts channel on startup

mrsimpson committed:
translations added from Lingohub (prior to next release)

mrsimpson committed:
minor corrections for translations

mrsimpson committed:
Check confirmation value on chect close

ruKurz committed:
Merge pull request #5 from assistify/hotfix/#2-and-more-translations

Hotfix/#2 and more translations

ruKurz committed:
Merge pull request #4 from assistify/hotfix/#3-cancel-close

Check confirmation value on chat close

mrsimpson committed:
Create subscriptions with all-notification (#9)

* Create subscriptions with all-notification for email, mobile and Desktop on creation of a request

* correction subscribe to all Desktop-Notifications

mrsimpson committed:
Handle duplicate request channel number after renaming (#12)

mrsimpson committed:
enable serch for expertises and requests

mrsimpson committed:
enable channel settings for custom room types

mrsimpson committed:
enable members list for custom room types

mrsimpson committed:
satisfy es-lint

mrsimpson committed:
Merge branch 'master' into develop

Commit:
smarti#87 add support area to smarti

mrsimpson committed:
added option to specify creation template for custom room types

mrsimpson committed:
Merge pull request #62 from assistify/smarti#87_add_support_area_to_smarti_call

Add support-area to smarti

mrsimpson committed:
custom room types: prevent renaming

mrsimpson committed:
custom room types: include in global search

mrsimpson committed:
merge implementation from 0.58.2

mrsimpson committed:
Global announcement: showing a custom message in alert-color on login- and home-screen

mrsimpson committed:
Global announcement: Text and translation of setting

mrsimpson committed:
Global announcement: Text and translation of setting

mrsimpson committed:
Global announcement (#71)

* Global announcement: showing a custom message in alert-color on login- and home-screen

* Global announcement: Text and translation of setting

mrsimpson committed:
Merge branch 'core/#8264-room-types' into feature/#19-decouple-help-requests

# Conflicts:
#	.meteor/packages
#	.meteor/versions
#	packages/rocketchat-channel-settings/client/views/channelSettings.js
#	packages/rocketchat-channel-settings/server/functions/saveRoomName.js
#	packages/rocketchat-i18n/i18n/de.i18n.json
#	packages/rocketchat-ui-admin/client/rooms/adminRooms.js
#	packages/rocketchat-ui-flextab/client/tabs/membersList.js
#	packages/rocketchat-ui-message/client/messageBox.js
#	server/methods/eraseRoom.js
#	server/publications/spotlight.js

mrsimpson committed:
Make message box input auto grow on focus.
This re-sizes the height and adapts the button, e. g. when replying.

mrsimpson committed:
Merge branch 'hotfix/#63-global-announcement' into develop

mrsimpson committed:
Custom room type: handle not-provided creation template

mrsimpson committed:
Merge branch 'core/#8264-room-types' into feature/#19-decouple-help-requests

Commit:
disable smarti load if necessary parameters are not set

mrsimpson committed:
Update to new UI and room types API

mrsimpson committed:
Enable custom room type creation templates

mrsimpson committed:
Merge branch 'core/#8264-room-types' into feature/#19-decouple-help-requests

mrsimpson committed:
Provide wrapper for room creation for better extensibility

mrsimpson committed:
WIP: Provide new style creation templates - not functional yet, only graphically adapted

mrsimpson committed:
Merge branch 'core/#8264-room-types' into feature/#19-decouple-help-requests

mrsimpson committed:
Merge branch 'develop' of https://github.com/RocketChat/Rocket.Chat into core/#8264-room-types

mrsimpson committed:
Replace hard-coded room types with "interface"-method-calls

mrsimpson committed:
thx, codacy

mrsimpson committed:
Merge branch 'core/#8264-room-types' into feature/#19-decouple-help-requests

# Conflicts:
#	packages/rocketchat-livechat/app/i18n/de.i18n.json

mrsimpson committed:
Restored icons
Now based on SVG. Provisioned at startup using `meteor-inject-initial`
Icons created from font awesome using [icomoon](https://icomoon.io/app/#/select)

mrsimpson committed:
Tabbed user interface for custom room types creation
Addresses RocketChat/Rocket.Chat#8264

mrsimpson committed:
Tabbed user interface for custom room types creation
Addresses RocketChat/Rocket.Chat#8264

mrsimpson committed:
Add new UI paradigm creation dialog

mrsimpson committed:
Merge branch 'core/#8264-room-types-creation-ui' into feature/#19-decouple-help-requests

mrsimpson committed:
A bit of beautification: Use theme colors and a transition when switching between tabs

mrsimpson committed:
Merge branch 'core/#8264-room-types-creation-ui' into feature/#19-decouple-help-requests

mrsimpson committed:
fixes #10 - tested in chrome and IE on windows

mrsimpson committed:
remove dbs common, old templates for previous versions, local result handling

mrsimpson committed:
correct dependency

mrsimpson committed:
Import from meteor-package. However, it results in a runtime error

mrsimpson committed:
Merge remote-tracking branch 'origin/hotfix/#10-sidebar-IE-issue' into develop

mrsimpson committed:
Updated RC to 0.58.4

mrsimpson committed:
Fixes RocketChat#8446

- Configurable global announcement via settings
- Uses theme "attention-color" for styling (don't complain to me with respect to the default colors!)
- Appears on Login, Home and Channel-Screens
- Can be confirmed if a user is logged-in
- Button floats which allows long and short texts

mrsimpson committed:
Make global announcement confirmable - cherry-picked from PR RocketChat#8461

mrsimpson committed:
Merge branch 'master' into hotfix/#63-global-announcement

mrsimpson committed:
Merge pull request #90 from assistify/hotfix/#63-global-announcement

Make global announcement confirmable

graywolf336 committed:
Room definitions are more typed but extended classes have full control over their definitions beyound the expected.

mrsimpson committed:
Proper exporting and importing of dependencies

mrsimpson committed:
remove unnecessary addFiles - it's the mainModule

graywolf336 committed:
Move the settings configuration over to an ENUM based.
This allows for each different type of room to define
which setting it wants to display instead of them
being hard coded on the settings tab controller.

graywolf336 committed:
Fix the tests failing due to some items not being on all settings

mrsimpson committed:
merge latst version of room types API

mrsimpson committed:
Adapt custom room types to latest version of the (more object oriented) API

mrsimpson committed:
Merge remote-tracking branch 'origin/master' into develop

mrsimpson committed:
Rename file from merge - somehow, git wrecked things up

graywolf336 committed:
[DOC] Update the rocketchat:lib readme with information about providing custom room types

mrsimpson committed:
Fix creation of expertises

graywolf336 committed:
Fix the creation tab page and fix expertise auto complete

graywolf336 committed:
Fix the room name's not being displayed in the admin rooms

mrsimpson committed:
WIP: Create page for request - giving headache on autocomplete

mrsimpson committed:
Merge branch 'feature/#19-decouple-help-requests' of https://github.com/assistify/Rocket.Chat into feature/#19-decouple-help-requests

# Conflicts:
#	packages/assistify-help-request/client/views/sideNav/AssistifyCreateExpertise.js

graywolf336 committed:
Fix the popup for the requests

graywolf336 committed:
Remove the console.log from the userAutocomplete publish

mrsimpson committed:
Fix expertise autocomplete displaying the relevant results

mrsimpson committed:
Styling of knowledge base panel

mrsimpson committed:
Enable hiding of creation pages e. g. if the user is not authorized to create the room type

mrsimpson committed:
Add creation validation configuration, e. g. in order to check whether the user is authorized to create the room type

mrsimpson committed:
Merge branch 'core/#8264-room-types' into core/#8264-room-types-creation-ui

# Conflicts:
#	packages/rocketchat-lib/lib/roomTypesCommon.js

mrsimpson committed:
hide room-type-creation-dialog if room type cannot be created

mrsimpson committed:
Add proper authorizations for requests and topics

mrsimpson committed:
merge core/#8264-room-types - modified:   packages/rocketchat-ui/client/views/app/room.js

mrsimpson committed:
merge core/#8264-room-types-creation-ui - modified:   packages/rocketchat-theme/client/imports/general/variables.css

mrsimpson committed:
Merge branch 'feature/#19-decouple-help-requests' into release/0.59.0_rc15-0.4.0

mrsimpson committed:
fixes for rc15

mrsimpson committed:
merge feature/#82-cleanup packages

mrsimpson committed:
de-duplicate tabs for Assistify.AI

mrsimpson committed:
Merge branch 'core/#8446-global-announcement' into release/0.59.0_rc15-0.4.0

mrsimpson committed:
Hide default external search tabbar (in live chat)

mrsimpson committed:
Do not cache added find methods

mrsimpson committed:
Refactoring of packages

mrsimpson committed:
More positive request description

mrsimpson committed:
add smarti invalidation and support area

mrsimpson committed:
Satisfy linters and spelling-addicts

mrsimpson committed:
merge 0.59.0

mrsimpson committed:
Fix autocomplete for expertises

mrsimpson committed:
Fix: AI-tab did not appear when entering a request or live-chat-room anymore

mrsimpson committed:
Restored icon in knowledge-base-tab

mrsimpson committed:
Add beta-tag - fixes #94

mrsimpson committed:
Fix Smarti widget for RC 0.59

sampaiodiego committed:
Fix guest pool inquiry taking

mrsimpson committed:
Merge tag '0.59.1' into release/0.59.0-0.4.0

mrsimpson committed:
remove redundant knowledge base icon

mrsimpson committed:
Make favorites and unread rooms appear on top

mrsimpson committed:
Shorten texts for creation screen in order to allow mobile usage

mrsimpson committed:
Denote members of a topic as experts

mrsimpson committed:
allow scrolling in creation dialog

mrsimpson committed:
Typo

mrsimpson committed:
Missing translation (create-request-button)

mrsimpson committed:
Typo

mrsimpson committed:
Adapt warning-text for leaving a room. Fixes #92

mrsimpson committed:
Adapt warning-text for leaving a room. Fixes #92

mrsimpson committed:
Fixes Smarti widget trying to connect very eagerly. Fixes #112

mrsimpson committed:
Fix UI-unit-tests

mrsimpson committed:
merge 0.59.2. conflict in packages/rocketchat-ui/client/views/app/room.js should be properly resolved

mrsimpson committed:
Fix missing global declaration from merge 0.59.2

mrsimpson committed:
Move profile in membersList to interface method

mrsimpson committed:
merge latest version of room types api

vickyokrm committed:
quick hack for IE

mrsimpson committed:
Merge pull request #114 from assistify/hotfix/#10-sidebar-IE-issue

quick hack for IE

mrsimpson committed:
Better UX for unavailable Smarti

mrsimpson committed:
Merge branch 'release/0.59.0-0.4.0' of https://github.com/assistify/Rocket.Chat into release/0.59.2-0.4.2

mrsimpson committed:
Hide double-scrollbars in Smarti tab

mrsimpson committed:
Try: make smarti-content scrollable again

mrsimpson committed:
Make deletion button visible again - fixes #117

mrsimpson committed:
Fix AI tab not opening automatically anymore

mrsimpson committed:
Propagate CORS-enabling to Restivus - fixes #7915

mrsimpson committed:
next try hiding the second scrollbar in Smarti

mrsimpson committed:
Merge pull request #98 from assistify/release/0.59.0-0.4.0

Release 0.59.2-0.4.2 - massive refactoring, new RC base version, all new and almost shiny

mrsimpson committed:
Merge pull request #119 from assistify/release/0.59.0-0.4.0

Release 0.59.2-0.4.2

mrsimpson committed:
Don't remove the subscription on closing, only hide the room

mrsimpson committed:
Merge pull request #121 from assistify/hotfix/#120-close-request

Needs manual testing with multiple parties =>deploy to test

vickyokrm committed:
fixes the request channel title display issue

mrsimpson committed:
Merge pull request #130 from assistify/hotfix/#109-room-title-display-issue

fixes the request channel title display issue

mrsimpson committed:
add historic chats icon - fixes 8703 (#124)

(cherry picked from commit 13776a5)

mrsimpson committed:
#116 close request (#122)

* Don't remove the subscription on closing, only hide the room

* Add interface method for UI-texts.
Provide implementation for the confirmation dialogs on leave and hide

(cherry picked from commit a480c20)

* Added a closing text constant

(cherry picked from commit c5c0ece)

* Closing a request adds the comment properly and creates an automated message explaining what that means - fixes #120

* fixed typo

* Fix frontend-exception once a room has been left

* Message that indicates a closed request room #116
- improved messages

ruKurz committed:
add extended logging information when loading smarti js #136

mrsimpson committed:
Merge pull request #137 from assistify/136-extennd-logginng

add extended logging information when loading smarti js #136

mrsimpson committed:
Merge branch 'master' into develop

# Conflicts:
#	packages/assistify-help-request/server/methods/closeHelpRequest.js

mrsimpson committed:
Merge tag '0.59.3' into develop

# Conflicts:
#	.docker/Dockerfile
#	.sandstorm/sandstorm-pkgdef.capnp
#	.travis/snap.sh
#	HISTORY.md
#	package.json
#	packages/rocketchat-ldap/server/ldap.js
#	packages/rocketchat-ldap/server/sync.js
#	packages/rocketchat-lib/rocketchat.info
#	packages/rocketchat-ui/client/views/app/room.js

mrsimpson committed:
Log with a defined log level

mrsimpson committed:
Allow room types to provide an own "no subscriptions yet" text

mrsimpson committed:
Provide meaningful "no subscription yet"-texts - fixes #133

mrsimpson committed:
Smarti adapter buffer invalidation on settings change

mrsimpson committed:
Minor wording update - fixes #138

Commit:
Fixed Issues 87, 88

mrsimpson committed:
Feature/#140 disabled user message (#142)

* Provide proper error handling on failed login attempts 
* Prevent password reset for disabled users
* Propagate information about being disabled to the user - fixes #140

mrsimpson committed:
more robust publication of help requests

mrsimpson committed:
Bump version to 0.4.4

mrsimpson committed:
Merge branch 'release/0.59.3-0.4.4'

mrsimpson committed:
Merge branch 'release/0.59.3-0.4.4' into develop

sampaiodiego commented:
hi @HenrikScholl , can you please describe what does your PR do? thx

mrsimpson committed:
Fix incorrectly cased Systemlogger leading to backend exceptions

mrsimpson committed:
Merge pull request #150 from assistify/hotfix/undefined-reference-SystemLogger

Hotfix: Undefined reference Systemlogger

vickyokrm committed:
Feature/#23 title first message to new request (#149)

* Request title and first message while creating new request
* Make titles of inputs and placeholders more consistent
* Fix display issues:

- Dropdown overlapped by input field
- In English, the width of the creation dialog was not 100%, thus input fields within too narrow

* Minor corrections with respect to error handling:

- Refactor error display to an own template
- Show error if selected expertise on request creation is invalid (not chosen from the dropdown)
- Fix positioning of "at" on members selection if invalid

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=8847) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=8847) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: mrsimpson<br/>:x: vickyokrm<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/Rocket.Chat?pullRequest=8847) it.</sub>

ruKurz committed:
Fixes #151 - Misspelled label "jetzt chaten" (#152)

mrsimpson committed:
Corrections to creation dialog (#154)

* Corrections to creation dialog

- Propagate first message properly (fixes #153)
- Change "Name" to "Title"
- get rid of flashing error message on auto-complete-confirmation with tab

* Corrections to creation dialog

- more robust error handling
- prevent flashing of validation errors

* Fix improper clearing of request title

* Invalid Expertise field highlight

mrsimpson committed:
#156 Prefill request creation form (#157)

* fixes #156

- Pass topic as URL param (URL encoded) as `topic` or `expertise` (same effect)
- Title is focused if expertise is passed

* Allow pre-filling of title and question as well.

This way, fixes #20 : The consumer (e. g. the wiki page) can create a simple form an pass the content as parameter

* - Use the copied current() instead of the internal value
```
FlowRouter.current
ƒ () {                                                                               // 248
  // We can't trust outside, that's why we clone this                                                        …
```
- refactor query-selectors => reuse

* use FlowRouter API properly

mrsimpson committed:
Merge branch 'master' into develop

mrsimpson committed:
Merge branch 'develop' of https://github.com/assistify/Rocket.Chat into develop

mrsimpson committed:
Setting based permissions - downport (#158)

* Allow maintenance of per-setting permissions

(cherry picked from commit eed869a)

* Implicitly assign and revoke setting group permissions

(cherry picked from commit 28b769b)

* Improve Display of setting permissions

(cherry picked from commit 8523456)

* Add path to permission title

(cherry picked from commit c87a30d)

* Permission to access setting permissions

(cherry picked from commit 48b1076)

* Adapt wording

(cherry picked from commit daccad8)

* UI-adaptation: Allow users with permission 'manage-selected-permissions' to see and change the affected settings.
However, this is not reactive: Once the permissions for a particular setting are changed, the user needs to log  off and on again before it becomes effective in the UI.
This is most probably a consequence of the CachedCollection. This collection needed to be changed on permission-change.
In the backend however, the permissions become effective immediately.

(cherry picked from commit 00e4bb5)

* Don't adapt sorting on the client side

(cherry picked from commit 9b71b62)

* Fix: Apply changed setting permissions reactively

(cherry picked from commit 293ad73)

* Move setting-based permissions to own collection

(cherry picked from commit 8f59f1c)

* Unify collections for setting and other permissions again into one

(cherry picked from commit 8d923c2)

* Get rid of frontend exceptions on changing selected settings

(cherry picked from commit a7fdc87)

* - Sort permissions by group
- Do not try to create permissions for hidden settings in higher-level-callbacks
- Remove `setting-permissions` collection - fully integrated into `permissions`

(cherry picked from commit f007231)

* Harmonize wording in German

(cherry picked from commit 5cf5df2)

mrsimpson committed:
Merge pull request #162 from assistify/feature/livechat-integration

Travis failed due to timeout => all checks have passed.

mrsimpson committed:
German language informalized (#160)

* German language informalized - Liebe Deutsche, wir kennen euch nun besser. Wir wollen ab jetzt “Du” zu einander sagen 😉

* Update de.i18n.json

* Update de.i18n.json

* Update de.i18n.json

mrsimpson committed:
Allow administration even if user has got only `edit-privileged-setting` but not `view-privileged-setting`

mrsimpson committed:
Revert "Fixed Issues 87, 88 (livechat on mobile devices)" (#164)

mrsimpson committed:
Create configuration expert role on startup (#159)

* Allow maintenance of per-setting permissions

(cherry picked from commit eed869a)

* Implicitly assign and revoke setting group permissions

(cherry picked from commit 28b769b)

* Improve Display of setting permissions

(cherry picked from commit 8523456)

* Add path to permission title

(cherry picked from commit c87a30d)

* Permission to access setting permissions

(cherry picked from commit 48b1076)

* Adapt wording

(cherry picked from commit daccad8)

* UI-adaptation: Allow users with permission 'manage-selected-permissions' to see and change the affected settings.
However, this is not reactive: Once the permissions for a particular setting are changed, the user needs to log  off and on again before it becomes effective in the UI.
This is most probably a consequence of the CachedCollection. This collection needed to be changed on permission-change.
In the backend however, the permissions become effective immediately.

(cherry picked from commit 00e4bb5)

* Don't adapt sorting on the client side

(cherry picked from commit 9b71b62)

* Fix: Apply changed setting permissions reactively

(cherry picked from commit 293ad73)

* Move setting-based permissions to own collection

(cherry picked from commit 8f59f1c)

* Unify collections for setting and other permissions again into one

(cherry picked from commit 8d923c2)

* Get rid of frontend exceptions on changing selected settings

(cherry picked from commit a7fdc87)

* - Sort permissions by group
- Do not try to create permissions for hidden settings in higher-level-callbacks
- Remove `setting-permissions` collection - fully integrated into `permissions`

(cherry picked from commit f007231)

* Harmonize wording in German

(cherry picked from commit 5cf5df2)

* add configuration package

* Add default role configuration on startup

* set default system language to DE

* Reduce capabilities of config expert and introduce minor admin

* Parted the roles for configuration and managing the rest

- Manager - well - manages the application, like a minor admin. Target is that this role is capable of doing everything which is necessary while *regularly* running the application
- Config-expert is allowed to customize the application (affecting all users' experience)

mrsimpson committed:
Informal german language for our custom texts (#165)

* German language informalized - Liebe Deutsche, wir kennen euch nun besser. Wir wollen ab jetzt “Du” zu einander sagen 😉

* German texts of custom enhancement informalized

mrsimpson committed:
Bump version to 0.5.0

mrsimpson committed:
Merge branch 'release/v0.59.3-0.5.0' into develop

mrsimpson committed:
Smarti 0.6.1 (#147)

* WIP: Proxy for Smarti stub

* WIP: Proxy for Smarti stub added to package

```
curl -X POST \
  http://localhost:3000/api/v1/assistify/smarti/conversation/ \
  -H 'content-type: application/json' \
  -H 'x-auth-token: EKZ2_t7Wfqg5GML8bs44v3uS0d_b7f3rUd9IUYKNpvJ' \  //from login
  -H 'x-user-id: S4Mew2PDtEPr3Ejne' \                               //from login
  -d '{
	"thisIsMy":"JSON-Body"
}'
```

* use injected logger for logging in API

* Fixes #151 - Misspelled label "jetzt chaten"

* Using the configured access token as HTTP header

* Using the RC proxy for Smarti

* Revert "Feature/#23 title first message to new request (#149)"

This reverts commit 484b04cc461af0da7df1d7103de3ba8109e3ec61.

* Revert "Fixes #151 - Misspelled label "jetzt chaten""

This reverts commit 0c9ac4f0f25a5863ac55f4c9256ff24ea07769c5.

* Fixing lint errors.

* Fixing non authorized access.

* fixed issue with wrong knowledge provider indexes

* improved tailing slashes for URLs

* - add authorization checks before routing the API calls
- roll back injected logger, since it is not defined

* removed unused localization keys

* Reducing code for adding a tailing slash to the Smarti URL.

* consolidate constants naming

* - Use RateLimiter for Smarti requests
- Use propagation function for 'onMessage' and 'onClose'
- Make Smarti the default knowledge provider
- Reordered Smarti backend settings
- Added descriptions for Smarti backend settings
- Also reload the settings, when reloading the Smarti widget
- Refactored file names

* Using rate limiter in each proxy method and limit the propagateToSmarti function instead of HTTP.call

* - merged proxy and adapter into only one file
- only use DDP for Smarti Widget / Rocket.Chat messages (getConversation, getSmartiQueryBuilderResult)
- added proxy endpoint for Smarti conversation search
- added several comments

* Migrate settings

* Extract Smarti loader

* additional migrations

* Do not migrate old settings with other defaults

* separated responsibilities

* Simplify adapter, proxy, router, widgetBackend

* Minor fixes to get it work with Smarti.

* Fix Webhook-token not being transmitted

* revert unintentional changes to other files after branching off the wron state.
Copied all changes from `develop` which were not included in `assistify:ai`-package

mrsimpson committed:
Avoid `upsert` in migration script

mrsimpson committed:
Make inserting new renamed settings more robust
Create new config prior to renaming settings
